05-read-write-plot.Rmd

```diff
@@ -1,62 +1,167 @@
 # Geographic data I/O {#read-write}
 
+## Prerequisites {-}
+
+- This chapter requires the packages **tidyverse**, **sf** and **raster**. 
+
+```{r, message=FALSE}
+library(tidyverse)
+library(sf)
+library(raster)
+```
+
+## Introduction
 The previous chapters provided an overview of spatial data classes in R, with a focus on simple features and rasters.
 This chapter is about getting spatial data onto your computer and then, perhaps after processing it with techniques described in this book, back out to the world.
-Geocomputation usually involves reading existing data, and saving it after processing sections.
-Therefore, it is important be aware of the most often used spatial file format and their strong and weak sides (see section \@ref(file-formats)).
+Before we can even think about processing data, we need to retrieve it, i.e. we need to know which sources are at our disposal (section \@{#retrieving-data}).
+Having data, we need to know how we can import them in our software.
+<!-- so far, I haven't seen a discussion on the pros and cons -->
+Therefore, it is important to be aware of the most often used spatial file formats and their pros and cons (see section \@ref(file-formats)).
 For the same purposes knowing how to efficiently read and write spatial data is crucial (see sections \@ref(data-input) and \@ref(data-output)).
 We also include a section (\@ref(visual-outputs)) on visualization because outputting data in a human (not just computer) readable format enables non-programmers to benefit from your work.
-If your aim is to use geocomputation to improve the world, e.g. by encouraging evidence-based policies, this final stage is vital.
 
 I/O is short for "input/output" which means, in plain English, "reading and writing data".
-We use the acronym instead of plain English not to confuse you or to make chapter names short, but because that's the term used in computer science and it is useful to think of data import and export from a computing perspective.^[
+We use the acronym instead of plain English not to confuse you or to make chapter names short, but because that is the term used in computer science, and it is useful to think of data import and export from a computing perspective.^[
 Concepts such as computational efficiency, hard disk space and 'idempotence' are useful when thinking about reading and writing geographic datasets, which can become large and difficult to handle.
 Loading/saving data is yet another way of saying the same thing.
 ]
 
-## File formats
+## Retrieving open data {#retrieving-data}
```

Comment:
I would start with retrieving data since this is the natural first step before doing anything else such as importing it.